tools/db_stress.cc

```diff
@@ -419,6 +419,10 @@ DEFINE_int32(acquire_snapshot_one_in, 0,
              "If non-zero, then acquires a snapshot once every N operations on "
              "average.");
 
+DEFINE_bool(compare_full_db_state_snapshot, false,
```

Comment:
we should see if there's a way to turn this on in our CI tests without reducing coverage of the other features too much. But we can do it later.